typing/includecore.ml

```diff
@@ -378,6 +390,16 @@ let report_private_object_mismatch env ppf err =
   | Missing s -> pr "The implementation is missing the method %s" s
   | Types err -> report_type_inequality env ppf err
 
+let report_kind_mismatch ppf first second kind1 kind2 =
+  let pr fmt = Format.fprintf ppf fmt in
+  let kind_to_string = function
+  | Kind_abstract -> "abstract"
+  | Kind_record -> "a record"
+  | Kind_variant -> "a variant"
+  | Kind_open -> "an extensible variant" in
+  pr "Their kinds differ: %s is %s, but %s is %s"
```

Comment:
This is a good occasion to remove the never explained notion of `kind` from the error message.
```
This is a variant, but the original is an extensible variant
```
is as informative as
```
Their kind differ: this is a variant, but the original is an extensible variant
```

Reply:
Cool, I've removed "Their kinds differ"!